### PR TITLE
Users microservice implementation

### LINKED ISSUE #1: Create user functional
IN: **user credentials(username & password)**
OUT: **status**

### LINKED ISSUE #2: Validate user functional
IN: **user credentials**
OUT: **userId**

### LINKED ISSUE #3: Delete user functional
IN: **userId**(from header)
OUT: **status**

### LINKED ISSUE #4: Create and connect UsersDB
Screenshot 2024-03-01 at 14 26 26

**postgres**

### PR DESCRIPTION
**Pull request by MaksimPegov**

### CODE REVIEW

**Comment by MaksimPegov** [comment]
Microservice was copied from todosss-api and now it is running and working with db. Some fixes and unnecessary code refactoring required.

File: docker-compose.yml
Comment (antonpegov) [inline docker-compose.yml:1]: Looks like you forgot to remove this file

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.postgres
```
Comment (antonpegov) [inline .gitignore]: Why we need this? We should't have any local database.

File: _db/create_db_user.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (antonpegov) [inline _db/create_db_user.sh]: Why we need this file here? Is it for local database?

File: eureka-server/target/eureka-server-1.0-SNAPSHOT.jar
Comment (antonpegov) [inline eureka-server/target/eureka-server-1.0-SNAPSHOT.jar:1]: I beleave we can add this to .gitignore

File: eureka-server/target/maven-archiver/pom.properties
Comment (antonpegov) [inline eureka-server/target/maven-archiver/pom.properties:1]: Do we need this in the repo?

File: eureka-server/target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst
Comment (antonpegov) [inline eureka-server/target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst:1]: Do we need this *.lst files in the repository?

File: users/target/classes/com/maksimpegov/users/UsersApplication.class
Comment (antonpegov) [inline users/target/classes/com/maksimpegov/users/UsersApplication.class:1]: Can we add *.class to .gitignore?

**Review by antonpegov — COMMENTED** [review]
Have same questions. But looks great!

File: docker-compose.yml
Comment (MaksimPegov) [reply docker-compose.yml:1]: yes, i will remove it in next update

File: .gitignore (hunk 1 above)
Comment (MaksimPegov) [reply .gitignore]: This is volume of database, it will be stored in _db folder in next update

File: _db/create_db_user.sh (hunk 2 above)
Comment (MaksimPegov) [reply _db/create_db_user.sh]: This is temp file for testing, i will remove it later

File: eureka-server/target/eureka-server-1.0-SNAPSHOT.jar
Comment (MaksimPegov) [reply eureka-server/target/eureka-server-1.0-SNAPSHOT.jar:1]: Ok

File: eureka-server/target/maven-archiver/pom.properties
Comment (MaksimPegov) [reply eureka-server/target/maven-archiver/pom.properties:1]: i don't know, i already removed it in a new version

File: eureka-server/target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst
Comment (MaksimPegov) [reply eureka-server/target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst:1]: this is some system staf for eureka-server, it is generated automatically by eureka-server

File: users/target/classes/com/maksimpegov/users/UsersApplication.class
Comment (MaksimPegov) [reply users/target/classes/com/maksimpegov/users/UsersApplication.class:1]: OK!

File: docker-compose.yml
Comment (antonpegov) [reply docker-compose.yml:1]: Pls create task for it to not forget

**Pull request merged**